functions/Add-DbaComputerCertificate.ps1

```diff
@@ -78,36 +75,39 @@ Adds the local C:\temp\cer.cer to the local computer's LocalMachine\My (Personal
 			return
 		}
 		
-		if (!(Test-Path -Path $path)) {
-			Write-Message -Level Warning -Message "$Path does not exist"
-			return
+		if ($Path) {
+			if (!(Test-Path -Path $Path)) {
+				Write-Message -Level Warning -Message "$Path does not exist"
+				return
+			}
 		}
 		
-		foreach ($computer in $computername) {
-			if (![dbavalidate]::IsLocalhost($computer) -and !$Password) {
-				Write-Message -Level Output -Message "You have specified a remote computer. A password is required for private key encryption/decryption for import."
-				$Password = Read-Host -AsSecureString -Prompt "Password"
+		if (![dbavalidate]::IsLocalhost($computer) -and !$Password) {
+			$Password = ((65 .. 90) + (97 .. 122) | Get-Random -Count 29 | ForEach-Object { [char]$_ }) -join "" | ConvertTo-SecureString -AsPlainText -Force
+		}
+		
+		if ($path) {
+			try {
+				# This may be too much, but ¯\_(ツ)_/¯
+				$bytes = [System.IO.File]::ReadAllBytes($path)
+				$Certificate = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2
+				$Certificate.Import($bytes, $password, [System.Security.Cryptography.X509Certificates.X509KeyStorageFlags]::DefaultKeySet)
 			}
-			
-			if ($Certificate) {
-				$tempdir = ([System.IO.Path]::GetTempPath()).TrimEnd("\")
-				$tempfile = "$tempdir\cert.cer"
-				$certdata = $certificate.Export([System.Security.Cryptography.X509Certificates.X509ContentType]::PFX, $password)
+			catch {
+				Stop-Function -Message "Can't import certificate." -ErrorRecord $_ -Continue
```

Comment:
`-Continue` parameter should only be used within a loop.
Given general advice probably a non-issue though in this instance